include/libtorrent/announce_entry.hpp

```diff
@@ -37,12 +37,41 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "libtorrent/time.hpp"
 #include "libtorrent/error_code.hpp"
 #include "libtorrent/string_view.hpp"
+#include "libtorrent/address.hpp"
 
 #include <string>
 #include <cstdint>
+#include <vector>
 
 namespace libtorrent
 {
+	namespace aux { struct session_listen_socket; }
+
+	struct TORRENT_EXPORT announce_endpoint
```

Comment:
since this is a public type, would you mind adding a comment documenting it, above this declaration? The documentation tool will pick this up